src/client/components/layouts/Header/Header.test.tsx

```diff
@@ -319,17 +319,18 @@ describe('Header', () => {
 				expect(screen.getByText('Je deviens mentor')).toBeInTheDocument();
 			});
 
-			it('positionne le menu dans le bon sous menu de niveau 2 et permet de retourner en arrière', () => {
+			it('positionne le menu dans le bon sous menu de niveau 2 et permet de retourner en arrière', async () => {
+				const user = userEvent.setup();
```

Comment:
Pourquoi ne plus passer par fireEvent ? (ça fait un moment que je me pose la question de pourquoi on utilise les 2 sur le projet ça aurait été bien d'uniformiser)

Reply:
En fait `userEvent` c'est la version _user-centric_ de `fireEvent` qui est JS-centric. Par exemple, quand tu fais un `fireEvent.change`, ce qui est fait derrière en gros c'est : 
```js
// fireEvent.change(input, {target: {value: '2020-05-24'}})
input.dispatchEvent('change', new Event(/* ... */));
```
Quand tu fais `userEvent.type`, ce qu'il fait c'est : 
```js
// userEvent.type(input, '2020-05-24')
input.dispatchEvent('click', new Event());
input.dispatchEvent('change', new Event(/* ... */);
```
qui est beaucoup plus proche du comportement réel d'un utilisateur.

Aussi, vu que `userEvent` est _user-centric_, la syntaxe est beaucoup plus simple : 
```js
userEvent.type(input, 'toto');
// Je tape dans le champ "input" le mot "toto"

fireEvent.change(input, { target: { value: 'toto' } })
// Je dispatch un event sur l'élément "input" dont la value est "toto"
```

Enfin c'est beaucoup moins lié à l'implémentation : 
```js
fireEvent.change(input, { target: { value: 'toto' } });
// Le test passe jusqu'à ce qu'au lieu d'utiliser `value`, j'utilise `currentValue` dans le composant ce qui ne change pas le comportement dans la plupart des cas
// Le test passe jusqu'à ce que ce soit plus `onChange` mais `onKeyPress` par exemple
// Le test passe alors que le composant casse si je fais des effets de bord avec d'autres propriété de l'event (genre `preventDefault`, `stopPropagation`, toucher à la valeur dans le _handler_, ...)
```

Reply:
btw il y a une petite note en haut de la [page de doc de `fireEvent`](https://testing-library.com/docs/dom-testing-library/api-events/) qui dit de préférer `userEvent` : 
> Note
Most projects have a few use cases for fireEvent, but the majority of the time you should probably use [@testing-library/user-event](https://testing-library.com/docs/user-event/intro).

Reply:
cool merci pour le partage de connaissance, ça vaudra le coup de faire un petit ADR la dessus au prochain jardinage tech. On s'est déjà posé la question de quoi utiliser avec Doro :)

Reply:
C'est noté dans l'agenda